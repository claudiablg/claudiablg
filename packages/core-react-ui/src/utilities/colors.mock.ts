export const generatedColorCSSVarNames = [
  '--color-transparent',
  '--color-primary-100',
  '--color-primary-300',
  '--color-primary-500',
  '--color-primary-700',
  '--color-primary-900',
  '--color-accent1-100',
  '--color-accent1-300',
  '--color-accent1-500',
  '--color-accent1-700',
  '--color-accent1-900',
  '--color-accent2-100',
  '--color-accent2-300',
  '--color-accent2-500',
  '--color-accent2-700',
  '--color-accent2-900',
  '--color-accent3-100',
  '--color-accent3-300',
  '--color-accent3-500',
  '--color-accent3-700',
  '--color-accent3-900',
  '--color-grey-0',
  '--color-grey-25',
  '--color-grey-50',
  '--color-grey-100',
  '--color-grey-200',
  '--color-grey-300',
  '--color-grey-400',
  '--color-grey-500',
  '--color-grey-600',
  '--color-grey-700',
  '--color-grey-800',
  '--color-grey-900',
  '--color-grey-1000',
  '--color-grey-0-reversed',
  '--color-grey-100-reversed',
  '--color-grey-200-reversed',
  '--color-effect-transparent-light',
  '--color-effect-transparent-medium',
  '--color-effect-transparent-heavy',
  '--color-utility-green-100',
  '--color-utility-green-300',
  '--color-utility-green-500',
  '--color-utility-green-700',
  '--color-utility-green-900',
  '--color-utility-yellow-100',
  '--color-utility-yellow-300',
  '--color-utility-yellow-500',
  '--color-utility-yellow-700',
  '--color-utility-yellow-900',
  '--color-utility-red-100',
  '--color-utility-red-300',
  '--color-utility-red-500',
  '--color-utility-red-700',
  '--color-utility-red-900',
  '--color-intent-accessibility-color',
  '--color-intent-transparent',
  '--color-intent-primary',
  '--color-intent-primary-reversed',
  '--color-intent-secondary',
  '--color-intent-secondary-reversed',
  '--color-intent-primary-text',
  '--color-intent-primary-text-reversed',
  '--color-intent-secondary-text',
  '--color-intent-secondary-text-reversed',
  '--color-intent-tertiary-text',
  '--color-intent-tertiary-text-reversed',
  '--color-intent-success-text',
  '--color-intent-success-action',
  '--color-intent-success-background',
  '--color-intent-warning-text',
  '--color-intent-warning-action',
  '--color-intent-warning-background',
  '--color-intent-danger-text',
  '--color-intent-danger-action',
  '--color-intent-danger-background',
  '--color-intent-background0',
  '--color-intent-background1',
  '--color-intent-background2',
];
