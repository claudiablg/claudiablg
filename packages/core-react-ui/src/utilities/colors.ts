import { Color } from '@newrade/core-design-system';

export function getHSLColor({ h, s, l, a }: Partial<Color> = { h: 200, s: 100, l: 100, a: 1 }) {
  return `hsla(${h}, ${s}%, ${l}%, ${a || 1})`;
}

export function lightenColor(color: Color, amount: number) {}

export function darkenColor(color: Color, amount: number) {}
