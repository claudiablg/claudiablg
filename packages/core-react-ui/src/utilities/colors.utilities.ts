import { Color } from '@newrade/core-design-system';

export function getHSLColor({ h, s, l, a }: Partial<Color>) {
  return `hsl(${h}deg ${s}% ${l}% / ${a || 100}%)`;
  // return `hsla(${h}, ${s}%, ${l}%, ${a || 1})`;
}

export function lightenColor(color: Color, amount: number) {}

export function darkenColor(color: Color, amount: number) {}
