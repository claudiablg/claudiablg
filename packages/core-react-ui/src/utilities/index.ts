export * from './colors.utilities';
export * from './component.utilities';
export * from './css-variable.utilities';
export * from './effects.utilities';
export * from './filter-color.utility';
export * from './text.utilities';
export * from './theme.utilities';
export * from './utilities';
