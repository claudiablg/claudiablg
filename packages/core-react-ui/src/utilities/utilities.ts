import { ColorIntents, Colors } from '@newrade/core-design-system';

/**
 * Create default color intents.
 */
export function createDefaultColorIntents(colors: Colors): ColorIntents {
  return {
    primary: colors.primary500,
    primaryReversed: colors.greyscale0,
    secondary: colors.primary500,
    secondaryReversed: colors.greyscale0,
    primaryText: colors.greyscale900,
    primaryTextReversed: colors.greyscale0,
    secondaryText: colors.greyscale700,
    secondaryTextReversed: colors.greyscale0,
    tertiaryText: colors.greyscale600,
    tertiaryTextReversed: colors.greyscale0,
    successText: colors.utilityGreen1000,
    successAction: colors.utilityGreen500,
    successBackground: colors.utilityGreen100,
    warningText: colors.utilityYellow1000,
    warningAction: colors.utilityYellow500,
    warningBackground: colors.utilityYellow100,
    dangerText: colors.utilityRed1000,
    dangerAction: colors.utilityRed500,
    dangerBackground: colors.utilityRed100,
    background0: colors.greyscale0,
    background1: colors.greyscale100,
    background2: colors.greyscale200,
  };
}

/**
 * Convert a number (in px) into a rem value.
 * @param px value in pixel
 * @param baseUnitPx font-size set on the <html/> element
 */
export function pxToRem({ baseUnitPx, value }: { baseUnitPx: number; value: number }): string {
  return `${value / baseUnitPx}rem`;
}

/**
 * Convert a number (in px) into a em relative value.
 * @param px value in pixel
 * @param fontSize font-size set on the element
 */
export function pxToEm({ fontSize, value }: { fontSize: number; value: number }): string {
  return `${value / fontSize}em`;
}

/**
 * Convert px value (e.g. '15px') into a rem value.
 * @param px value in string px value
 * @param baseUnitPx font-size set on the <html/> element
 */
export function pxStringToRem({ baseFontSize, value }: { baseFontSize: number; value: string }): string {
  if (!value?.length) {
    return '';
  }

  const pxPattern = /(\d+\.?(\d+))/g;
  const match = pxPattern.exec(value);
  if (!(match?.length && match[0]?.length)) {
    return '';
  }
  const conversion = Number(match[0]);
  if (!conversion) {
    return '';
  }

  return `${conversion / baseFontSize}rem`;
}

/**
 * Convert px value (e.g. '15px') into a number value.
 * @param px value in string px value
 */
export function pxStringToNumber({ value }: { value: string }): number | undefined {
  if (!value?.length) {
    return undefined;
  }

  const pxPattern = /(\d+\.?(\d+))/g;
  const match = pxPattern.exec(value);
  if (!(match?.length && match[0]?.length)) {
    return undefined;
  }
  const conversion = Number(match[0]);
  if (conversion === undefined || conversion === NaN || conversion === null) {
    return undefined;
  }

  return conversion;
}
