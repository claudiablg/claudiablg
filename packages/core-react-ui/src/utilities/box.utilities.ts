import { Border, BorderStyles, BoxStyle, Radius } from '@newrade/core-design-system';
import { getCSSColor } from './colors.utilities';
import { keys, px } from './utilities';

export function getCSSBoxStyle(box: BoxStyle): BoxStyle<string> {
  return {
    width: px({ value: box.width }),
    height: px({ value: box.height }),
    // padding: box.padding,
    border: getCSSBorderStyles(box.border),
    // outline: box.outline,
    // backgroundColor: box.backgroundColor,
  };
}

export function getCSSBorderStyles(borders?: BorderStyles): BorderStyles<string> | undefined {
  if (!borders) {
    return undefined;
  }

  return keys(borders).reduce((previous, current) => {
    return { ...previous, [current]: getCSSBoxBorder(borders[current]) };
  }, {} as BorderStyles<string>);
}

export function getCSSBoxBorder(border?: Border): Border<string> {
  if (!border) {
    return {};
  }

  return {
    color: border.color ? getCSSColor(border.color) : undefined,
    width: border.width ? px({ value: border.width }) : undefined,
    style: border.style || 'solid',
    radius: getCSSBoxRadius(border.radius),
  };
}

/**
 * @see https://developer.mozilla.org/en-US/docs/Web/CSS/border-radius
 */
export function getCSSBoxRadius(radius?: Radius): string {
  if (!radius) {
    return '';
  }

  return radius.all
    ? `${radius.all}`
    : `${radius.topLeft} ${radius.topRight} ${radius.bottomRight} ${radius.bottomLeft}`;
}
