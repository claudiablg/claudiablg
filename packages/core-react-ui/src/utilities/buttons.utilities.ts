import * as DS from '@newrade/core-design-system';
import { ButtonColors, ButtonSizes, ButtonVariants, DesignSystem } from '@newrade/core-design-system';
import { kebab } from 'case';
import { CSSButtons, CSSButtonsVarNames } from '../design-system/css-buttons';
import { getCSSBoxStyle } from './box.utilities';
import { getCSSColor, getCSSVarForColors, getCSSVarNamesForColors } from './colors.utilities';
import { keys } from './utilities';

export function getCSSButtons(theme: DesignSystem): CSSButtons {
  const { colors, components } = theme;
  const { buttons } = components;

  const colorVar = getCSSVarForColors({
    colors: colors.colors,
    colorIntents: colors.colorIntents,
  });
  const colorVarNames = getCSSVarNamesForColors({
    colors: colors.colors,
    colorIntents: colors.colorIntents,
  });

  const buttonsVar = getCSSVarNameForButtons({
    buttons,
    prefix: 'btn',
    varBrackets: true,
  });
  const buttonsVarNames = getCSSVarNameForButtons({
    buttons,
    prefix: 'btn',
    varBrackets: false,
  });

  return {
    var: buttonsVar,
    varNames: buttonsVarNames,
    variants: getCSSButtonVariants(buttons.variants),
    sizes: getCSSButtonSizes(buttons.sizes),
  };
}

/**
 * Creates css variables from Buttons properties
 */
function getCSSVarNameForButtons({
  buttons,
  prefix,
  varBrackets,
}: {
  buttons: DS.Buttons;
  prefix: string;
  varBrackets: boolean;
}): CSSButtonsVarNames {
  const buttonsVarNames = {
    sizes: {},
    variants: {},
  } as CSSButtonsVarNames;
  const { variants, sizes } = buttons;

  keys(variants).forEach((variant) => {
    const currentVariant = variant as keyof CSSButtons['variants'];
    const formattedCurrentVariant = kebab(currentVariant)
      .replace('primary', 'prim')
      .replace('secondary', 'sec')
      .replace('tertiary', 'ter');
    if (currentVariant && currentVariant.length) {
      // handle variant
      if (variants[currentVariant] && typeof variants[currentVariant] === 'object') {
        const buttonColors = variants[currentVariant] as ButtonColors;
        keys(buttonColors).forEach((prop) => {
          const formattedPropName = kebab(prop);
          // @ts-ignore
          if (!buttonsVarNames['variants'][variant]) {
            // @ts-ignore
            buttonsVarNames['variants'][variant] = {};
          }
          if (varBrackets) {
            // @ts-ignore
            buttonsVarNames['variants'][variant][
              prop
            ] = `var(--${prefix}-${formattedCurrentVariant}-${formattedPropName})`;
            return;
          }

          // @ts-ignore
          buttonsVarNames['variants'][variant][prop] = `--${prefix}-${formattedCurrentVariant}-${formattedPropName}`;
          return;
        });
      }
    }
  });

  keys(sizes).forEach((variant) => {
    const currentVariant = variant as keyof CSSButtons['variants'];
    const formattedCurrentVariant = kebab(currentVariant)
      .replace('primary', 'prim')
      .replace('secondary', 'sec')
      .replace('tertiary', 'ter');
    if (currentVariant && currentVariant.length) {
      // handle variant
      if (variants[currentVariant] && typeof variants[currentVariant] === 'object') {
        const buttonColors = variants[currentVariant] as ButtonColors;
        keys(buttonColors).forEach((prop) => {
          const formattedPropName = kebab(prop);
          // @ts-ignore
          if (!buttonsVarNames['sizes'][variant]) {
            // @ts-ignore
            buttonsVarNames['sizes'][variant] = {};
          }
          if (varBrackets) {
            // @ts-ignore
            buttonsVarNames['sizes'][variant][
              prop
            ] = `var(--${prefix}-${formattedCurrentVariant}-${formattedPropName})`;
            return;
          }

          // @ts-ignore
          buttonsVarNames['sizes'][variant][prop] = `--${prefix}-${formattedCurrentVariant}-${formattedPropName}`;
          return;
        });
      }
    }
  });

  return buttonsVarNames;
}

function getCSSButtonVariants(buttons: ButtonVariants): ButtonVariants<string> {
  return keys(buttons).reduce((previous, current) => {
    const variantColorStyles: ButtonColors<string> = {
      backgroundColor: getCSSColor(buttons[current].backgroundColor),
    };
    previous = {
      ...previous,
      [current]: variantColorStyles,
    };
    return previous;
  }, {} as ButtonVariants<string>);
}

function getCSSButtonSizes(buttons: ButtonSizes): ButtonSizes<string> {
  return keys(buttons).reduce((previous, current) => {
    previous = {
      ...previous,
      [current]: getCSSBoxStyle(buttons[current]),
    };
    return previous;
  }, {} as ButtonSizes<string>);
}
