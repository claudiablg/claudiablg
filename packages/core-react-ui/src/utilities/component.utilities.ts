import { pascal } from 'case';
import { keys } from './utilities';

/**
 * <Link /> => "Link Medium Underline"
 */
export function getDefaultTextFromProps(name: string, props: object): string {
  const propString = `${keys(props)
    .map((prop) => pascal(props[prop]))
    .join(' ')}`;

  return `${pascal(name)} ${propString}`;
}
