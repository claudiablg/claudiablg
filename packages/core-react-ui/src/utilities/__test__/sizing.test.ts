import { Sizing } from '@newrade/core-design-system';
import { defaultSizing } from '../../design-system/default-sizing';
import { getCSSSizing } from '../sizing.utilities';

describe('shadows utilities', () => {
  describe(`${getCSSSizing.name}`, () => {
    it('should create a valid CSS box-shadow property', () => {
      const sizing = getCSSSizing(defaultSizing) as Sizing<string>;
      expect(sizing).toMatchSnapshot();
    });
  });
});
