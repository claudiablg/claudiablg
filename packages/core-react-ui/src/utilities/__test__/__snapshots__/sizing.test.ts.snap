// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shadows utilities getCSSSizing should create a valid CSS box-shadow property 1`] = `
Object {
  "baseFontSize": "9px",
  "ratio": 1.618,
  "sizes": Object {
    "desktop": Object {
      "x0": "9px",
      "x1": "9px",
      "x10": "144px",
      "x2": "13px",
      "x3": "21px",
      "x4": "34px",
      "x5": "55px",
      "x6": "89px",
      "x7": "44px",
      "x8": "33px",
      "x9": "77px",
    },
    "mobile": Object {
      "x0": "9px",
      "x1": "9px",
      "x10": "144px",
      "x2": "13px",
      "x3": "21px",
      "x4": "34px",
      "x5": "55px",
      "x6": "89px",
      "x7": "44px",
      "x8": "33px",
      "x9": "77px",
    },
    "tablet": Object {
      "x0": "9px",
      "x1": "9px",
      "x10": "144px",
      "x2": "13px",
      "x3": "21px",
      "x4": "34px",
      "x5": "55px",
      "x6": "89px",
      "x7": "44px",
      "x8": "33px",
      "x9": "77px",
    },
  },
  "var": Object {
    "x0": "var(-sizing-x0)",
    "x1": "var(-sizing-x1)",
    "x10": "var(-sizing-x10)",
    "x2": "var(-sizing-x2)",
    "x3": "var(-sizing-x3)",
    "x4": "var(-sizing-x4)",
    "x5": "var(-sizing-x5)",
    "x6": "var(-sizing-x6)",
    "x7": "var(-sizing-x7)",
    "x8": "var(-sizing-x8)",
    "x9": "var(-sizing-x9)",
  },
  "varNames": Object {
    "x0": "--sizing-x0",
    "x1": "--sizing-x1",
    "x10": "--sizing-x10",
    "x2": "--sizing-x2",
    "x3": "--sizing-x3",
    "x4": "--sizing-x4",
    "x5": "--sizing-x5",
    "x6": "--sizing-x6",
    "x7": "--sizing-x7",
    "x8": "--sizing-x8",
    "x9": "--sizing-x9",
  },
}
`;
